Assets/Scripts/UnityServices/Lobbies/LobbyServiceFacade.cs

```diff
@@ -77,88 +81,45 @@ public void BeginTracking()
+                if (m_LocalUser.IsHost)
                 {
-                    if (m_LocalUser.IsHost)
-                    {
-                        task = DeleteLobbyAsync(lobbyId);
-                    }
-                    else
-                    {
-                        task = LeaveLobbyAsync(lobbyId);
-                    }
+                    await DeleteLobbyAsync();
+                }
+                else
+                {
+                    await LeaveLobbyAsync();
                 }
-
-                m_LocalUser.ResetState();
-                m_LocalLobby?.Reset(m_LocalUser);
             }
 
             if (m_IsTracking)
             {
-                m_UpdateRunner.Unsubscribe(UpdateLobby);
                 m_IsTracking = false;
-                m_HeartbeatTime = 0;
-                m_JoinedLobbyContentHeartbeat.EndTracking();
-            }
-
-            return task;
-        }
-
-        async void UpdateLobby(float unused)
-        {
-            if (!m_RateLimitQuery.CanCall)
-            {
-                return;
-            }
-
-            try
-            {
-                var lobby = await m_LobbyApiInterface.GetLobby(m_LocalLobby.LobbyID);
-
-                CurrentUnityLobby = lobby;
-                m_LocalLobby.ApplyRemoteData(lobby);
-
-                // as client, check if host is still in lobby
-                if (!m_LocalUser.IsHost)
+                if (m_LobbyEvents != null && m_LobbyEventConnectionState != LobbyEventConnectionState.Unsubscribed)
                 {
-                    foreach (var lobbyUser in m_LocalLobby.LobbyUsers)
+                    try
                     {
-                        if (lobbyUser.Value.IsHost)
-                        {
-                            return;
-                        }
+                        await m_LobbyEvents.UnsubscribeAsync();
                     }
-                    m_UnityServiceErrorMessagePub.Publish(new UnityServiceErrorMessage("Host left the lobby", "Disconnecting.", UnityServiceErrorMessage.Service.Lobby));
-                    await EndTracking();
-                    // no need to disconnect Netcode, it should already be handled by Netcode's callback to disconnect
-                }
-            }
-            catch (LobbyServiceException e)
-            {
-                if (e.Reason == LobbyExceptionReason.RateLimited)
-                {
-                    m_RateLimitQuery.PutOnCooldown();
-                }
-                else if (e.Reason != LobbyExceptionReason.LobbyNotFound && !m_LocalUser.IsHost) // If Lobby is not found and if we are not the host, it has already been deleted. No need to publish the error here.
-                {
-                    PublishError(e);
+                    catch (ObjectDisposedException e)
+                    {
+                        // This exception occurs in the editor when exiting play mode without first leaving the lobby.

```

Comment:
only when exiting playmode or actually OnApplicationQuit which could happen in builds as well?

Reply:
Only when exiting play mode